php/src/Controller/DockerController.php

```diff
@@ -149,22 +149,32 @@ public function StopContainer(Request $request, Response $response, $args) : Res
     public function StartDomaincheckContainer() : void
     {
         # Don't start if domain is already set
-        if ($this->configurationManager->GetDomain() != '') {
+        if ($this->configurationManager->GetDomain() !== '' || $this->configurationManager->wasStartButtonClicked()) {
             return;
         }
 
         $id = 'nextcloud-aio-domaincheck';
 
-        $container = $this->containerDefinitionFetcher->GetContainerById($id);
-        // don't start if the domaincheck is already running
-        if ($container->GetIdentifier() === $id && $container->GetRunningState() instanceof RunningState) {
-            return;
-        // don't start if apache is already running
-        } elseif ($container->GetIdentifier() === self::TOP_CONTAINER && $container->GetRunningState() instanceof RunningState) {
+        $cacheKey = 'domaincheckWasStarted';
+
+        $domaincheckContainer = $this->containerDefinitionFetcher->GetContainerById($id);
+        $apacheContainer = $this->containerDefinitionFetcher->GetContainerById(self::TOP_CONTAINER);
+        // Don't start if apache is already running
+        if ($apacheContainer->GetRunningState() instanceof RunningState) {
             return;
+        // Don't start if domaincheck is already running
+        } elseif ($domaincheckContainer->GetRunningState() instanceof RunningState) {
+            $domaincheckWasStarted = apcu_fetch($cacheKey);
+            // Start domaincheck again when 10 minutes are over by not returning here
+            if($domaincheckWasStarted !== false && is_string($domaincheckWasStarted)) {
```

Comment:
```suggestion
            if ($domaincheckWasStarted !== false && is_string($domaincheckWasStarted)) {
```

This seems to be the style used in this project :) Might be a good idea to add php-cs-fixer at some point in this repo ;)

Reply:
thanks for the review! will do that at some point ;)